subsys/bluetooth/host/iso.c

```diff
@@ -346,7 +344,7 @@ void bt_iso_connected(struct bt_conn *iso)
 	if (bt_iso_setup_data_path(iso)) {
 		BT_ERR("Unable to setup data path");
 #if defined(CONFIG_BT_ISO_BROADCAST)
-		if (iso->iso.is_bis) {
+		if (iso->iso.type != BT_ISO_CHAN_TYPE_CONNECTED) {
```

Comment:
Use explicit check similar to in line 310, and assertion on BT_ISO_CHAN_TYPE_NONE

```suggestion
		if (iso->iso.type == BT_ISO_CHAN_TYPE_BROADCASTER ||
		    iso->iso.type == BT_ISO_CHAN_TYPE_SYNC_RECEIVER) {
```

Reply:
Considered doing that, but didn't as that would change unnecessary many lines, but it is a good idea to do it as it is no longer a boolean value. Will update.